src/main/java/org/opensearch/sdk/ssl/DefaultSslKeyStore.java

```diff
@@ -0,0 +1,834 @@
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import javax.crypto.Cipher;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.SSLParameters;
+
+import io.netty.handler.codec.http2.Http2SecurityUtil;
+import io.netty.handler.ssl.ApplicationProtocolConfig;
+import io.netty.handler.ssl.ApplicationProtocolConfig.Protocol;
+import io.netty.handler.ssl.ApplicationProtocolConfig.SelectedListenerFailureBehavior;
+import io.netty.handler.ssl.ApplicationProtocolConfig.SelectorFailureBehavior;
+import io.netty.handler.ssl.ApplicationProtocolNames;
+import io.netty.handler.ssl.ClientAuth;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.SslProvider;
+import io.netty.handler.ssl.SupportedCipherSuiteFilter;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+//import org.bouncycastle.asn1.ASN1InputStream;
+//import org.bouncycastle.asn1.ASN1ObjectIdentifier;
+//import org.bouncycastle.asn1.ASN1Primitive;
+//import org.bouncycastle.asn1.ASN1Sequence;
+//import org.bouncycastle.asn1.ASN1String;
+//import org.bouncycastle.asn1.ASN1TaggedObject;
+
+import org.opensearch.OpenSearchException;
+import org.opensearch.OpenSearchSecurityException;
+import org.opensearch.SpecialPermission;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.env.Environment;
+import org.opensearch.sdk.ssl.util.CertFileProps;
+import org.opensearch.sdk.ssl.util.CertFromFile;
+import org.opensearch.sdk.ssl.util.CertFromKeystore;
+import org.opensearch.sdk.ssl.util.CertFromTruststore;
+import org.opensearch.sdk.ssl.util.ExceptionUtils;
+import org.opensearch.sdk.ssl.util.KeystoreProps;
+import org.opensearch.transport.NettyAllocator;
+
+import static org.opensearch.sdk.ssl.SecureSSLSettings.SSLSetting.SSL_TRANSPORT_CLIENT_KEYSTORE_KEYPASSWORD;
+import static org.opensearch.sdk.ssl.SecureSSLSettings.SSLSetting.SSL_TRANSPORT_CLIENT_PEMKEY_PASSWORD;
+import static org.opensearch.sdk.ssl.SecureSSLSettings.SSLSetting.SSL_TRANSPORT_KEYSTORE_KEYPASSWORD;
+import static org.opensearch.sdk.ssl.SecureSSLSettings.SSLSetting.SSL_TRANSPORT_KEYSTORE_PASSWORD;
+import static org.opensearch.sdk.ssl.SecureSSLSettings.SSLSetting.SSL_TRANSPORT_PEMKEY_PASSWORD;
+import static org.opensearch.sdk.ssl.SecureSSLSettings.SSLSetting.SSL_TRANSPORT_SERVER_KEYSTORE_KEYPASSWORD;
+import static org.opensearch.sdk.ssl.SecureSSLSettings.SSLSetting.SSL_TRANSPORT_SERVER_PEMKEY_PASSWORD;
+import static org.opensearch.sdk.ssl.SecureSSLSettings.SSLSetting.SSL_TRANSPORT_TRUSTSTORE_PASSWORD;
+
+public class DefaultSslKeyStore implements SslKeyStore {
+
+    private static final String DEFAULT_STORE_TYPE = "JKS";
+
+    private void printJCEWarnings() {
+        try {
+            final int aesMaxKeyLength = Cipher.getMaxAllowedKeyLength("AES");
+
```

Comment:
I see most of the changes are from security repo[1]. It would be hard to maintain this in multiple places. 
Can these classes be packaged as a library and SDK can consume them?

[1] https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/ssl/DefaultSecurityKeyStore.java

Reply:
That begs the question: where would be the best location to store it?

These files are not identical because I changed the constants to remove `plugins.security` and also removed any reference to `HTTP` since extension nodes are not HTTP servers. I agree it should be extracted, but to where?

Would that block this PR, what do you think about filing an issue to abstract the common code into a library that both repos can consume?

Reply:
Sure definitely lets track this in an issue, I dont want to block our progress. 
This is operational pain but incremental progress is important :)